### DISCUSSION TITLE
Redesign of the Time & Date spoke

### DISCUSSION THREAD

**Discussion by poncovka** [rhinstaller/anaconda#5355] [Ideas]
The team would like to remove the timezone map from the Time & Date spoke in the GTK-based user interface (GUI). It would solve some issues that we have with the map and its dependencies.  However, the spoke doesn't look good without the map, so we would like to do a small redesign as well. @mairin Would you be able to give us some pointers?

We did some research of the time and date configuration for the new user interface (Web UI) and it looked like there is a general tendency to move away from the map in the system configuration tools. For example, https://gitlab.gnome.org/Teams/Design/settings-mockups/-/issues/23.

<details>
<summary>The current look of the spoke</summary>

![timezone_map_current](https://github.com/rhinstaller/anaconda/assets/7396092/ef1a0ea8-29dc-40a4-b355-35a3f927ca04)

</details>
<details>
<summary>The spoke without the timezone map</summary>

![timezone_map_removed](https://github.com/rhinstaller/anaconda/assets/7396092/9786824f-188f-4b77-8309-f4e6c78c04b7)

</details>

**Comment by mairin** [comment]
![Board](https://github.com/rhinstaller/anaconda/assets/799683/f413977d-6c9a-4c3c-bfd4-0271369acd99)

Here is one idea! This would eliminate the NTP configuration being a separate pop up dialog since we have the real estate the map occupied free to use now.

Let me know what you think.

**Reply by jkonecny12** [reply]
Hi, I like it. Just a note, we have a requirement to enable configuration of 24 / 12(AM/PM) format even for NTP. So I guess it could be resolved with this redesign.

**Reply by jstodola** [reply]
https://issues.redhat.com/browse/RHEL-4781

**Reply by poncovka** [reply]
I don't understand the request. How does it suppose to work? The time is controlled by chrony with NTP enabled.

**Reply by poncovka** [reply]
Oh, I see. It is only for the AM/PM configuration. Does the customer understand that this configuration doesn't have any impact on the installed system? We use it to display the manual time in the installer. That's all.

**Reply by jkonecny12** [reply]
Interesting, I thought it's written to the system too. Seems like there is confusion raised from the design.

**Comment by mairin** [comment]
Hey! Thanks for the feedback. Here is an updated mockup; the changelog:

- Time zone & time format have been pulled out of the two selection options and apply across the board.
- The NTP UI components are removed and will be available via the "Configure NTP ..." button akin to how it works today.
- Manual date & time has had the time format selection removed; however, if the user is in AM/PM mode, an AM/PM toggle will appear for setting the time manually. If the user is in 24-hour mode, the AM/PM toggle will disappear.

Let me know what you think!

![Anaconda-5355-v2](https://github.com/rhinstaller/anaconda/assets/799683/28bef6cf-d65d-4e34-9954-cc6ba27e51a2)

**Reply by jkonecny12** [reply]
This looks great to me. Thanks a lot!

**Reply by jkonecny12** [reply]
On the second look it's left aligned. From looking on Anaconda spokes we have Subscription, Root and User spokes center aligned instead. It feels inconsistent to me.

**Comment by mairin** [comment]
Here's an updated version with the content center-aligned. Changelog:

- broke time format out into its own row to accommodate center alignment not looking awkward :)
- changed section header casing to match existing screens (first word, first letter caps only; instead of first letter of each word capitalized)

![Anaconda-5355-v3](https://github.com/rhinstaller/anaconda/assets/799683/518ef4e7-a259-4886-992a-7cd84ed68c02)

as always let me know if there are any other issues :)

**Reply by jkonecny12** [reply]
Looks good to me. Thanks!

**Comment by mairin** [comment]
Here's another update, based on feedback around the time format selection. I misunderstood and thought the requirement was to provide time format selection that would persist, but the current selector only applies to manual date & time setting. 

Changelog:
- Time format row removed
- Radio button for 24h vs am/pm time format mobed to beside the manual time selection toggles

![anaconda-5355-v4](https://github.com/rhinstaller/anaconda/assets/799683/c5ebf68d-4a3d-44b3-bb96-f50183c9d202)

**Reply by poncovka** [reply]
It looks great :-) Thank you!

**Reply by jkonecny12** [reply]
I agree, thanks for the work!

**Comment by poncovka** [comment]
The implementation of the redesign is available at https://github.com/rhinstaller/anaconda/pull/5404.
